src/core/agent/services/ipexCommunicationService.ts

```diff
@@ -68,6 +68,29 @@ class IpexCommunicationService extends AgentService {
   @OnlineOnly
   async acceptAcdc(id: string): Promise<void> {
     const grantNoteRecord = await this.getNotificationRecordById(id);
+    const notificationRecord = await this.notificationStorage.findById(
+      grantNoteRecord.id
+    );
+
+    if (!notificationRecord) {
+      throw new Error(
+        `${IpexCommunicationService.NOTIFICATION_NOT_FOUND} ${id}`
+      );
+    }
+
+    if (
+      notificationRecord?.id &&
+      notificationRecord?.linkedGroupRequests &&
```

Comment:
I think we should stop adding `?` where it's not needed - `notificationRecord` is defined. Otherwise we might not notice bugs where something is meant to be defined